operator-framework/src/main/java/io/javaoperatorsdk/operator/api/ResourceController.java

```diff
@@ -28,4 +28,10 @@
      */
     UpdateControl<R> createOrUpdateResource(R resource, Context<R> context);
 
+    // What about we let developer to set the finalizer Name through overriding
+    // this method rather than setting it as attribute in annotation?
+    default String getDefaultFinalizerName() {
```

Comment:
Several places in the code finalizer references are named `defaultFinalzer`. Any objection to renaming them all to `finalizer`? I suppose they are indeed referring to the finalizer and not the default one. 
e.g. 
Several spots in this file:
https://github.com/java-operator-sdk/java-operator-sdk/blob/master/operator-framework/src/main/java/io/javaoperatorsdk/operator/processing/EventDispatcher.java#L24

Reply:
I agree.. sounds like we're calling everything defaultFinalizer even though we just mean the finalizer. This is actually confusing as there a thing called the default finalizer name. @csviri can you confirm we're seeing this right?

Reply:
I renamed the default finalizer to finalizer or similar everywhere in docs, logs and code for now. 
LMK if you have any objection/thoughts about it.

Reply:
@psycho-ir looks good, yep it jus tfinalizer in most of the cases not default finalizer. 
I don't think we should use default method in the interface however. Would stick to the annotation to override the finalizer name.

Reply:
Right, I have removed this method already.